src/tools/singlejar/test_util.cc

```diff
@@ -87,7 +108,7 @@ string GetEntryContents(const string &zip_path, const string &entry_name) {
   string command;
   blaze_util::StringPrintf(&command, "unzip -p %s %s", zip_path.c_str(),
                            entry_name.c_str());
-  FILE *fp = popen(command.c_str(), "r");
+  FILE *fp = popen(command.c_str(), "rb");
```

Comment:
Unfortunately this change breaks the internal version of `output_jar_simple_test`.
Is this change necessary?

Reply:
http://man7.org/linux/man-pages/man3/popen.3.html doesn't document `b`; it seems to be Microsoft-specific.

Reply:
The `"b"` is necessary because otherwise the Microsoft implementation will strip `\r\n` to `\n` and make many tests fail. I will add a `_WIN32` guard for this.

Reply:
Thanks!
With your permission, I'd like to also add this comment:
```
  // "b" is Microsoft-specific. It's necessary, because without it the popen
  // implementation will strip "\r\n" to "\n" and make many tests fail.
```
(If you agree to it, just let me know, no need to upload a new commit.)

Reply:
You have my permission.